src/compiler/parser/index.js

```diff
@@ -410,6 +416,31 @@ function processAttrs (el) {
   }
 }
 
+function checkVBindingVal (name: string, value: string): void {
+  let func
+
+  // there may be template error, leave it to the compiler detectErrors
+  try {
+    func = new Function(`(${value})`)
+  } catch (ex) {
+    return
+  }
+
+  try {
+    func()
```

Comment:
While being a savvy technique, this is also very very dangerous. It will introduce code execution, especially if expression is passed from user input or database.